src/components/tipRecords/TipRecordComponent.vue

```diff
@@ -51,23 +52,29 @@
   import FiatValueComponentVue from '../FiatValueComponent.vue';
   import util from '../../utils/util';
   import {EventBus} from '../../utils/eventBus';
+  import RetipComponent from '../RetipComponent.vue';
+
 
   export default {
     name: 'TipRecord',
     props: ['tip', 'foundWallet', 'fiatValue', 'senderLink'],
     components: {
-      'fiat-value': FiatValueComponentVue
+      'fiat-value': FiatValueComponentVue,
+      'retip-component': RetipComponent
     },
     data() {
       return {
+        showRetipWindow: false
       }
     },
     computed: {
       ...mapGetters(['current'])
     },
     methods: {
-      async retip(id) {
-        const amount = util.aeToAtoms(prompt("Tip Amount in AE?"));
+      toggleRetip(flag){
+        this.showRetipWindow = flag
+      },
+      async retip(id, amount) {
```

Comment:
why keep implementation of retip here and then pass it into the retip component?